Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM rust:1.45-buster as builder
+FROM rust:1.46-buster as builder
```

Comment:
this is an already huge PR. Do we need to add dep upgrades into it, or can that be done separately?

Reply:
The protobuf update was required due to a local build issue. This was addressed in https://github.com/mozilla-services/syncstorage-rs/pull/853. The rust image update brings our image closer to the currently stable rust branch.

Reply:
I get that the image update gets us closer to stable rust, but what I don't understand is why it's required as part of this work?

The protobuf update seems...completely unlreated? Unless I'm missing something obvious here.

Reply:
Basically, I can't develop, run or test on my machine unless I update protobuf. This is because protobuf is often a system level requirement and is not directly tied to a given project release. A single update of protobuf from anything will cause all versions of protobuf to update. It's possible to spin that out as [a dependency](https://github.com/mozilla-services/syncstorage-rs/pull/853), however it's often just more direct and quicker to deploy to roll it into an existing PR. 

As for the rust update, this makes sure that our current testing environment again matches the local development environment. This ensures that things like clippy, fmt and other tools operate the same and reduces the need for using CI as a live debugging environment to address issues that may only present themselves on one version.